src/tokenserver/db/models.rs

```diff
@@ -48,6 +50,13 @@ pub struct DbInner {
 }
 
 impl TokenserverDb {
+    // Note that this only works because an instance of `TokenserverDb` has *exclusive access* to
+    // a connection from the r2d2 pool for its lifetime. `LAST_INSERT_ID()` returns the ID of the
+    // most recently-inserted record *for a given connection*. If connections were shared across
+    // requests, using this function would introduce a race condition, as we could potentially
+    // get IDs from records created during other requests.
+    const LAST_INSERT_ID_QUERY: &'static str = "SELECT LAST_INSERT_ID() AS id";
```

Comment:
Just drawing attention to my comment here to make sure my logic checks out 😇

Reply:
I'll note that `LAST_INSERT_ID()` is problematic, but then so is damn near any other mechanism.

Reply:
Problematic enough not to use it?